ligo/stablecoin/actions.ligo

```diff
@@ -13,13 +13,101 @@
  * Helpers
  *)
 
+(*
+ * Helper function used to reduce
+ *  `if (condition) then failwith (message) else skip;`
+ * appearances.
+ *)
+function fail_on
+  ( const condition : bool
+  ; const message   : string
+  ) : unit is if condition then failwith (message) else unit
+
+(*
+ * Authorizes current contract owner and fails otherwise.
+ *)
+function authorize_owner
+  ( const store : storage
+  ) : unit is
+  fail_on
+    ( Tezos.sender =/= store.roles.owner
+    , "Sender is not owner"
+    )
+
+(*
+ * Ensures that sender is current pending owner
+ *)
+function authorize_pending_owner
+  ( const store : storage
+  ) : unit is case store.roles.pending_owner of
+    Some (pending_owner) -> fail_on
+      ( Tezos.sender =/= pending_owner
+      , "Sender is not current pending owner"
+      )
+    | None -> failwith ("No pending owners that await ownership transferring")
+    end
+
+(*
+ * Authorizes paused and fails otherwise.
+ *)
+function authorize_pauser
+  ( const store : storage
+  ) : unit is
+  fail_on
+    ( Tezos.sender =/= store.roles.pauser
+    , "Sender is not pauser"
+    )
+
+(*
+ * Authorizes master_minter and fails otherwise.
+ *)
+function authorize_master_minter
+  ( const store : storage
+  ) : unit is
+  fail_on
+    ( Tezos.sender =/= store.roles.master_minter
+    , "Sender is not master minter"
+    )
+
+(*
+ * Authorizes minter and fails otherwise.
+ *)
+function authorize_minter
+  ( const store : storage
+  ) : unit is case store.minting_allowances[Tezos.sender] of
+    Some (u) -> unit
+  | None -> failwith ("Sender is not minter")
+  end
+
+(*
+ * Helper that fails if the given contract is paused.
+ *)
+function ensure_not_paused
+  ( const store : storage
+  ) : unit is
+  fail_on
+    ( store.paused
+    , "Given operation is restricted during contract pause"
+    )
+
+(*
+ * Helper that fails if the given contract is paused.
```

Comment:
Should be 'is not paused' in think...

Reply:
@awkure The comments are still mixed up I think. For 'ensure_is_paused' it says 'fails if contract is paused' and for 'ensure_not_paused' it says 'fails if contract is not paused.'

Also the message for `ensure_is_paused` reads "Contract is paused", when it should be something like "contract should be paused, but isn't".

Reply:
@awkure This still needs to be fixed from the earlier comment.

>Also the message for ensure_is_paused reads "Contract is paused", when it should be something like "contract should be paused, but isn't".

Reply:
updated